### PR TITLE
Add Delete Comment Functionality

### LINKED ISSUE #261: FEATURE: Deletion functionality on comments
### Is there an existing issue raised for this?

- [X] I have searched the existing issues

### Describe the feature

This feature is regarding the deleting functionality on comments. If any user wants to delete its comment , then user can delete its comment.

### Component

I would like add this feature in  /api/recipe/delete-comment

### It should be implemented because

- Users may post incorrect or irrelevant comments (e.g., mistakes in ingredient suggestions or cooking tips). Having the ability to delete these ensures that the conversation around the recipe remains accurate and helpful.
- Deleting comments provides users control over their content. If they change their mind about a suggestion, or realize that their comment is redundant, they can remove it. This encourages engagement without the fear of permanent mistakes.

### Would you like to work on this issue?

Yes

### Terms & Conditions

- [X] I agree to follow the Code of Conduct
- [X] I'm a GSSOC'24 Ext contributor

### PR DESCRIPTION
**Pull request by Saurabhchaudhary9799**
## What does this PR do?
This PR implements the ability for users to delete comments on recipes. The following changes were made:
- Added a new handleDeleteComment function in the OneRecipe component.
- Integrated an API call to delete comments based on their ID.
- Updated the frontend to reflect the removal of a comment after it has been deleted.
- Ensured that only authorized users can delete comments by including a token in the request headers.

Fixes #261 

[Screencast from 2024-10-21 15-10-36.webm](https://github.com/user-attachments/assets/dc543764-699d-47f7-ae57-d6a1fa954f69)

## How should this be tested?

- Go to any recipe
- post any comment
- check that comment is being deleted or not

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Read Contributing guide
- [x] Self-reviewed my own code
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] The changes don't cause any responsiveness issues

### CODE REVIEW

**Comment by Saurabhchaudhary9799** [comment]
@AlfiyaSiddique , Please look into this PR.

File: .gitignore [hunk 1]
```diff
@@ -24,3 +24,4 @@ dist-ssr
 *.sw?
 
 /*/.env
+images
```
Comment (AlfiyaSiddique) [inline .gitignore:27]: @Saurabhchaudhary9799 why did you add images folder in .gitignore? IF it is for local setup please remove it

File: backend/server.js [hunk 2]
```diff
@@ -37,7 +37,7 @@ app.use(
     })
 );
 
-// app.use(cors({origin:"http://localhost:5173"})) // for local use
+
```
Comment (AlfiyaSiddique) [inline backend/server.js:40]: Do not remove this comment

File: frontend/src/App.jsx [hunk 3]
```diff
@@ -17,7 +17,7 @@ import AddRecipe from "./Pages/AddRecipe.jsx";
 import OneRecipe from "./Pages/OneRecipe.jsx";
 import UpdateRecipe from "./Pages/UpdateRecipe.jsx";
 import Contributors from "./Pages/Contributors.jsx"; // Import the Contributors component
-import PrivacyPolicy from "./Components/PrivacyPolicy.jsx";
+import LegalTerms from "./Components/LegalTerms.jsx";
 import NotFound from "./Pages/NotFound.jsx";
```
Comment (AlfiyaSiddique) [inline frontend/src/App.jsx:21]: @Saurabhchaudhary9799 why did you change the privacy policy component name to LegalTerms? If it is not for any valid reasons please rename it as it was before.

File: frontend/src/App.jsx [hunk 4]
```diff
@@ -79,9 +79,10 @@ function App() {
           <Route path="/user/:id/update/recipe" element={<UpdateRecipe />} />
           <Route path="/recipe/:id" element={<OneRecipe />} />
           <Route path="/contributors" element={<Contributors />} />
+          <Route path="/privacy-policy" element={<LegalTerms />} />
           <Route path="*" element={<NotFound/>}/>
           {/* Add Contributors route */}
-          <Route path="/privacy-policy" element={<PrivacyPolicy />} />
+
```
Comment (AlfiyaSiddique) [inline frontend/src/App.jsx:85]: Same comment as above

File: frontend/src/Components/LegalTerms.jsx [hunk 5]
```diff
@@ -70,4 +69,4 @@ const PrivacyPolicy = () => {
   );
 };
 
-export default PrivacyPolicy;
+export default LegalTerms;
```
Comment (AlfiyaSiddique) [inline frontend/src/Components/LegalTerms.jsx:72]: Please rename this file as it was before

File: frontend/src/Pages/OneRecipe.jsx [hunk 6]
```diff
@@ -1,71 +1,93 @@
...
+        console.error("Error fetching comments:", error);
+      }
+    };
+
+    fetchComments();
+  }, [recipe._id]);
+
+  // Function to handle posting a comment
+  // Function to handle posting a comment
+  const handlePostComment = async () => {
```
Comment (AlfiyaSiddique) [inline frontend/src/Pages/OneRecipe.jsx:38]: Duplicate comments: Remove one

**Comment by AlfiyaSiddique** [comment]
@Saurabhchaudhary9799 please improve the code.
Added some comments

File: frontend/src/App.jsx (hunk 3 above)
Comment (Saurabhchaudhary9799) [reply frontend/src/App.jsx:21]: @AlfiyaSiddique , I changed this name because there is an error generating by this . This name was changed for valid reasons.

File: frontend/src/App.jsx (hunk 4 above)
Comment (Saurabhchaudhary9799) [reply frontend/src/App.jsx:85]: @AlfiyaSiddique  ,  why do you want to revert these changes

**Comment by Saurabhchaudhary9799** [comment]
@AlfiyaSiddique , I did changes that you said , Please check once.

**Comment by AlfiyaSiddique** [comment]
@Saurabhchaudhary9799 The deployment preview is failing because you did not rename the legalterms component to privacy policy as it was before. Please refer to the master branch and rename it as it was.

**Comment by Saurabhchaudhary9799** [comment]
Screenshot from 2024-10-22 00-49-50

@AlfiyaSiddique ,  this is giving an error in deploying , so i changed  name again. But this  does not effect overall project

**Comment by AlfiyaSiddique** [comment]
@Saurabhchaudhary9799 did you change the legalterm file name to privacypolicy as it was before?? Please refer the master branch

**Comment by AlfiyaSiddique** [comment]
I can see there is a file named LegalTerms in the changed file section. You have renamed the PrivacyPolicy File  to LegalTerms unnecessarily. Please rename it as it is in Master branch

**Comment by Saurabhchaudhary9799** [comment]
> @Saurabhchaudhary9799 did you change the legalterm file name to privacypolicy as it was before?? Please refer the master branch

okay , i am doing it.

**Comment by Saurabhchaudhary9799** [comment]
@AlfiyaSiddique , i have done the change. Please check it.

**Comment by AlfiyaSiddique** [comment]
@Saurabhchaudhary9799 will be reviewed by tomorrow. Thanks for the quick change

**Comment by Saurabhchaudhary9799** [comment]
@AlfiyaSiddique , If everything is right , then kindly merge it.

File: .gitignore (hunk 1 above)
Comment (AlfiyaSiddique) [reply .gitignore:27]: @Saurabhchaudhary9799  ^^

File: backend/server.js (hunk 2 above)
Comment (AlfiyaSiddique) [reply backend/server.js:40]: @Saurabhchaudhary9799 ^^

File: frontend/src/Pages/OneRecipe.jsx (hunk 6 above)
Comment (AlfiyaSiddique) [reply frontend/src/Pages/OneRecipe.jsx:38]: @Saurabhchaudhary9799  ^^

**Comment by AlfiyaSiddique** [comment]
@Saurabhchaudhary9799 no issue with the deploy, it's successful,
I have added some comments to your code please fix them and update the PR. I will merge it then

**Comment by Saurabhchaudhary9799** [comment]
Screenshot from 2024-10-23 19-27-23
@AlfiyaSiddique , can you tell me how to resolve this conflicts

**Comment by AlfiyaSiddique** [comment]
@Saurabhchaudhary9799 use VS code merge conflict relsover editor to resolve.
Please refer this comment: https://github.com/AlfiyaSiddique/TastyTrails/pull/246#issuecomment-2420399301
